src/api/v4/dataview/category.js

```diff
@@ -0,0 +1,241 @@
+ * @param {carto.operation} options.operation - The operation to apply to the data.
+ * @param {string} options.operationColumn - The column name used in the operation.
+ *
+ * @constructor
+ * @extends carto.dataview.Base
+ * @memberof carto.dataview
+ * @api
+ */
+function Category (source, column, options) {
+  this._initialize(source, column, options);
+}
+
+Category.prototype = Object.create(Base.prototype);
+
+/**
+ * Set the dataview maxCategories
+ *
+ * @param  {string} maxCategories
+ * @return {carto.dataview.Category} this
+ * @api
+ */
+Category.prototype.setMaxCategories = function (maxCategories) {
+  this._checkMaxCategories(maxCategories);
+  this._options.maxCategories = maxCategories;
+  if (this._internalModel) {
+    this._internalModel.set('categories', maxCategories);
+  }
+  return this;
+};
+
+/**
+ * Return the current dataview maxCategories
+ *
+ * @return {string} Current dataview maxCategories
+ * @api
+ */
+Category.prototype.getMaxCategories = function () {
+  return this._options.maxCategories;
+};
+
+/**
+ * Set the dataview operation
+ *
+ * @param  {carto.operation} operation
+ * @return {carto.dataview.Category} this
+ * @api
+ */
+Category.prototype.setOperation = function (operation) {
+  this._checkOperation(operation);
+  this._options.operation = operation;
+  if (this._internalModel) {
+    this._internalModel.set('aggregation', operation);
+  }
+  return this;
+};
+
+/**
+ * Return the current dataview operation
+ *
+ * @return {carto.operation} Current dataview operation
+ * @api
+ */
+Category.prototype.getOperation = function () {
+  return this._options.operation;
+};
+
+/**
+ * Set the dataview operationColumn
+ *
+ * @param  {string} operationColumn
+ * @return {carto.dataview.Category} this
+ * @api
+ */
+Category.prototype.setOperationColumn = function (operationColumn) {
+  this._checkOperationColumn(operationColumn);
+  this._options.operationColumn = operationColumn;
+  if (this._internalModel) {
+    this._internalModel.set('aggregation_column', operationColumn);
+  }
+  return this;
+};
+
+/**
+ * Return the current dataview operationColumn
+ *
+ * @return {string} Current dataview operationColumn
+ * @api
+ */
+Category.prototype.getOperationColumn = function () {
+  return this._options.operationColumn;
+};
+
+/**
+ * Return the resulting data
+ *
+ * @return {CategoryData}
+ * @api
+ */
+Category.prototype.getData = function () {
+  if (this._internalModel) {
+    /**
+     * @typedef {object} CategoryItem
+     * @property {boolean} group
+     * @property {string} name
+     * @property {number} value
+     * @api
+     */
+    /**
+     * @typedef {object} CategoryData
+     * @property {number} count
+     * @property {number} max
+     * @property {number} min
+     * @property {number} nulls
+     * @property {string} operation
+     * @property {CategoryItem[]} result
+     * @property {string} type - Constant 'category'
```

Comment:
Do we need/want to expose this? I'd say we don't... In case we do, should we also expose a constant?

Reply:
Do you mean the `type` constants or the whole object documentation?

Reply:
Nop :)